dds2val/src/databroker.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+#################################################################################
+# Copyright (c) 2022 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License 2.0 which is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+#################################################################################
+
+import logging
+from typing import Any, Optional, Union
+
+import grpc.aio
+import kuksa_client.grpc
+from kuksa_client.grpc import (
+    DataEntry,
+    Datapoint,
+    DataType,
+    EntryUpdate,
+    Field,
+    Metadata,
+    VSSClient,
+)
+
+log = logging.getLogger(__name__)
+
+
+class Provider:
+    def __init__(
+        self, vss_client: VSSClient, grpc_metadata: Optional[grpc.aio.Metadata] = None
+    ):
+        self._name_to_type: dict[str, dict] = {}
+        self._rpc_kwargs = {"metadata": grpc_metadata}
+        log.info("Using %s", self._rpc_kwargs)
+        self._vss_client = vss_client
+
+    def register(
+        self, name: str, data_type: Union[str, DataType], description: str
+    ) -> bool:
+        if isinstance(data_type, str):
+            data_type = getattr(DataType, data_type)
+        try:
+            log.debug(
+                "register(%s, data_type: %s, '%s')",
+                name,
+                data_type.name,
+                description,
+            )
+            metadata = self._vss_client.get_metadata((name,), **self._rpc_kwargs)
+            if len(metadata) == 1:
+                self._name_to_type[name] = metadata[name].data_type
+                log.info(
+                    "%s was already registered with type %s",
+                    name,
+                    metadata[name].data_type.name,
+                )
+                return True
+
+            return self._register(name, data_type, description)
```

Comment:
This file is more or less the same as https://github.com/eclipse/kuksa.val.feeders/blob/main/dbc2val/dbcfeederlib/databroker.py (well actually the dbc2val version has been updated as we cannot register). 

I am not saying that we shall change it now as I think we want a bigger change, like having either this file as part of client code (in kuksa.val repo) or refactoring kuksa.val.feeders so that we get a "generic feeder module/package/lib" that can be used by all Python-based feeders. So maybe keep it as is and add it to a refactoring issue. There is anyway a need for refactoring in dbcfeeder as server/broker-specific code is not as hidden/encapsulated as I would like to.

Reply:
A little bit related to https://github.com/eclipse/kuksa.val.feeders/issues/62

Reply:
Will be tackled in another PR